### ISSUE TITLE
Not Streaming music - There was a problem with the requested skill's response

### ISSUE THREAD

**Issue by tinba12** [stevenleeg/geemusic#269]
I have been able to use Gee music to get information. For example If I say "ask gee music List the latest albums by The Wonder Years" it does work. If I say "Ask gee music Play music by A Tribe Called Quest" I get There was a problem with the requested skill's response. Response json is 

`{
	"body": {
		"version": "1.0",
		"response": {
			"outputSpeech": {
				"type": "PlainText",
				"text": "Playing top tracks by A Tribe Called Quest"
			},
			"card": {
				"type": "Standard",
				"title": "Playing top tracks by A Tribe Called Quest",
				"text": "",
				"image": {
					"smallImageUrl": "https://lh3.googleusercontent.com/muiawPgKoepzV5sjK5HxKcPzg1N34sGvqNLZIRisvwoa6dJ1pD_EbBVF6syo6f5Uh5rRxbR20A",
					"largeImageUrl": "https://lh3.googleusercontent.com/muiawPgKoepzV5sjK5HxKcPzg1N34sGvqNLZIRisvwoa6dJ1pD_EbBVF6syo6f5Uh5rRxbR20A"
				}
			},
			"directives": [
				{
					"type": "AudioPlayer.Play",
					"playBehavior": "REPLACE_ALL",
					"audioItem": {
						"stream": {
							"token": "7d1fc7a6-8040-4785-bbff-e76a28ebc6df",
							"url": "https:///<server>/alexa/stream/Txcshosrdbqfmim7fmh47f43yyi",
							"offsetInMilliseconds": 0
						}
					}
				}
			],
			"shouldEndSession": true
		},
		"sessionAttributes": {}
	}
}`

If I go to https:///<server>/alexa/stream/Txcshosrdbqfmim7fmh47f43yyi I get this error 

`gmusicapi.exceptions.CallFailure
gmusicapi.exceptions.CallFailure: GetStreamUrl: 403 Client Error: Forbidden for url: https://mclients.googleapis.com/music/mplay?opt=hi&net=mob&pt=e&slt=1542658057781&sig=ObGgrk-mSaIeidwMHRio3PsAoyk&mjck=Txcshosrdbqfmim7fmh47f43yyi&hl=en_US&dv=0&tier=aa
(requests kwargs: {'method': 'GET', 'url': 'https://mclients.googleapis.com/music/mplay', 'allow_redirects': False, 'headers': {'X-Device-ID': 'ECA86BFB2251', 'Authorization': '<omitted>'}, 'params': {'opt': 'hi', 'net': 'mob', 'pt': 'e', 'slt': '1542658057781', 'sig': b'ObGgrk-mSaIeidwMHRio3PsAoyk', 'mjck': 'Txcshosrdbqfmim7fmh47f43yyi', 'hl': 'en_US', 'dv': 0, 'tier': 'aa'}})`

**Comment by mvanholstyn** [comment]
@tinba12 did you get this working? I hit the same issue today.

**Comment by fergyfresh** [comment]
This is weird because this looks like a valid response.

```
{ "body": { "version": "1.0", "response": { "outputSpeech": { "type": "PlainText", "text": "Playing top tracks by A Tribe Called Quest" }, "card": { "type": "Standard", "title": "Playing top tracks by A Tribe Called Quest", "text": "", "image": { "smallImageUrl": "https://lh3.googleusercontent.com/muiawPgKoepzV5sjK5HxKcPzg1N34sGvqNLZIRisvwoa6dJ1pD_EbBVF6syo6f5Uh5rRxbR20A", "largeImageUrl": "https://lh3.googleusercontent.com/muiawPgKoepzV5sjK5HxKcPzg1N34sGvqNLZIRisvwoa6dJ1pD_EbBVF6syo6f5Uh5rRxbR20A" } }, "directives": [ { "type": "AudioPlayer.Play", "playBehavior": "REPLACE_ALL", "audioItem": { "stream": { "token": "7d1fc7a6-8040-4785-bbff-e76a28ebc6df", "url": "https:///<server>/alexa/stream/Txcshosrdbqfmim7fmh47f43yyi", "offsetInMilliseconds": 0 } } } ], "shouldEndSession": true }, "sessionAttributes": {} } }
```

**Comment by fergyfresh** [comment]
Can you try any of these other utterances that might work?

```

                "Play artist {artist_name}",
                "Play songs by {artist_name}",
                "Play top songs by {artist_name}",
                "Play top tracks by {artist_name}",
                "Play music by {artist_name}",
                "Play tracks by {artist_name}"
```

**Comment by tinba12** [comment]
I tried "ask gee music play artist lupe fiasco" I get the same error. The below is the response.
`{
	"body": {
		"version": "1.0",
		"response": {
			"outputSpeech": {
				"type": "PlainText",
				"text": "Playing top tracks by Lupe Fiasco"
			},
			"card": {
				"type": "Standard",
				"title": "Playing top tracks by Lupe Fiasco",
				"text": "",
				"image": {
					"smallImageUrl": "https://lh3.googleusercontent.com/TfO3gSHcOjshuPeEo9mLymYOh4gHAXx23Q1CA25K8Jth_RkH87qw6wfA87brDcj-9xEvn7xb",
					"largeImageUrl": "https://lh3.googleusercontent.com/TfO3gSHcOjshuPeEo9mLymYOh4gHAXx23Q1CA25K8Jth_RkH87qw6wfA87brDcj-9xEvn7xb"
				}
			},
			"directives": [
				{
					"type": "AudioPlayer.Play",
					"playBehavior": "REPLACE_ALL",
					"audioItem": {
						"stream": {
							"token": "cb8fa496-9ac5-43b5-be87-14fbe2cfb86a",
							"url": "https:///<Server>/alexa/stream/T2rzpdf6viys76x4q7z3luq4evy",
							"offsetInMilliseconds": 0
						}
					}
				}
			],
			"shouldEndSession": true
		},
		"sessionAttributes": {}
	}
}`

**Comment by fergyfresh** [comment]
Is that an error?

**Comment by tinba12** [comment]
Same error as above. There was a problem with the requested skill's response

**Comment by tinba12** [comment]
That is the server response to alexa.

**Comment by RickeyWard** [comment]
The problem is the stream url, that's what fails. I have the same issue. The Response from gee music is a valid json response, but when alexa tries to play the audio stream, that fetch fails with a 403 from google.

**Comment by RickeyWard** [comment]
You also get this message if you don't tell the alexa skill that it has "audio player" on the  interfaces.

**Comment by RickeyWard** [comment]
Though, that just gets you past the "there was a problem with the response" issue. It doesn't make your audio play.

`2018-12-01T23:10:41.046334+00:00 app[web.1]: 10.13.252.86 - - [01/Dec/2018 23:10:41] "GET /alexa/stream/Txrlu3maxeyxamtea24gpbvsj7a HTTP/1.1" 500 -
2018-12-01T23:10:41.046597+00:00 app[web.1]: Traceback (most recent call last):
2018-12-01T23:10:41.046599+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1997, in __call__
2018-12-01T23:10:41.046601+00:00 app[web.1]: return self.wsgi_app(environ, start_response)
2018-12-01T23:10:41.046602+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1985, in wsgi_app
2018-12-01T23:10:41.046603+00:00 app[web.1]: response = self.handle_exception(e)
2018-12-01T23:10:41.046605+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1540, in handle_exception
2018-12-01T23:10:41.046606+00:00 app[web.1]: reraise(exc_type, exc_value, tb)
2018-12-01T23:10:41.046607+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
2018-12-01T23:10:41.046609+00:00 app[web.1]: raise value
2018-12-01T23:10:41.046610+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
2018-12-01T23:10:41.046611+00:00 app[web.1]: response = self.full_dispatch_request()
2018-12-01T23:10:41.046612+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
2018-12-01T23:10:41.046614+00:00 app[web.1]: rv = self.handle_user_exception(e)
2018-12-01T23:10:41.046615+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
2018-12-01T23:10:41.046616+00:00 app[web.1]: reraise(exc_type, exc_value, tb)
2018-12-01T23:10:41.046617+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
2018-12-01T23:10:41.046618+00:00 app[web.1]: raise value
2018-12-01T23:10:41.046619+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
2018-12-01T23:10:41.046620+00:00 app[web.1]: rv = self.dispatch_request()
2018-12-01T23:10:41.046621+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
2018-12-01T23:10:41.046623+00:00 app[web.1]: return self.view_functions[rule.endpoint](**req.view_args)
2018-12-01T23:10:41.046624+00:00 app[web.1]: File "/app/geemusic/controllers.py", line 43, in redirect_to_stream
2018-12-01T23:10:41.046625+00:00 app[web.1]: stream_url = api.get_google_stream_url(song_id)
2018-12-01T23:10:41.046626+00:00 app[web.1]: File "/app/geemusic/utils/music.py", line 153, in get_google_stream_url
2018-12-01T23:10:41.046628+00:00 app[web.1]: return self._api.get_stream_url(song_id)
2018-12-01T23:10:41.046629+00:00 app[web.1]: File "<decorator-gen-102>", line 2, in get_stream_url
2018-12-01T23:10:41.046630+00:00 app[web.1]:
2018-12-01T23:10:41.046631+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gmusicapi/utils/utils.py", line 293, in wrapper
2018-12-01T23:10:41.046632+00:00 app[web.1]: return function(*args, **kw)
2018-12-01T23:10:41.046633+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gmusicapi/clients/mobileclient.py", line 388, in get_stream_url
2018-12-01T23:10:41.046635+00:00 app[web.1]: return self._make_call(mobileclient.GetStreamUrl, song_id, device_id, quality)
2018-12-01T23:10:41.046636+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gmusicapi/clients/shared.py", line 84, in _make_call
2018-12-01T23:10:41.046637+00:00 app[web.1]: return protocol.perform(self.session, self.validate, *args, **kwargs)
2018-12-01T23:10:41.046638+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gmusicapi/protocol/shared.py", line 226, in perform
2018-12-01T23:10:41.046639+00:00 app[web.1]: raise CallFailure(err_msg, call_name)
2018-12-01T23:10:41.046641+00:00 app[web.1]: gmusicapi.exceptions.CallFailure: GetStreamUrl: 403 Client Error: Forbidden for url: https://mclients.googleapis.com/music/mplay?opt=hi&net=mob&pt=e&slt=1543705840934&sig=OBN2w5wmgNDyUO5wRBnFcVMBbw4&mjck=Txrlu3maxeyxamtea24gpbvsj7a&hl=en_US&dv=0&tier=aa
2018-12-01T23:10:41.046642+00:00 app[web.1]: (requests kwargs: {'method': 'GET', 'url': 'https://mclients.googleapis.com/music/mplay', 'allow_redirects': False, 'headers': {'X-Device-ID': '661C0BF6C446', 'Authorization': '<omitted>'}, 'params': {'opt': 'hi', 'net': 'mob', 'pt': 'e', 'slt': '1543705840934', 'sig': b'OBN2w5wmgNDyUO5wRBnFcVMBbw4', 'mjck': 'Txrlu3maxeyxamtea24gpbvsj7a', 'hl': 'en_US', 'dv': 0, 'tier': 'aa'}})
2018-12-01T23:10:41.046644+00:00 app[web.1]: (response was: '<HTML>\n<HEAD>\n<TITLE>Forbidden</TITLE>\n</HEAD>\n<BODY BGCOLOR="#FFFFFF" TEXT="#000000">\n<H1>Forbidden</H1>\n<H2>Error 403</H2>\n</BODY>\n</HTML>\n')`

**Comment by fergyfresh** [comment]
I think you meant *not validated. Some things work with the gmusicapi and
some dont. There is work being done on the gmusicapi side to harden this a
bit so we wont have so many errors on our end.

**Comment by tinba12** [comment]
I have the audio interface enabled.  My Device id is valid.  When I say "ask gee music List the latest albums by The Wonder Years" I get proper response.

**Comment by fergyfresh** [comment]
You might have to use this https://www.myfakeinfo.com/mobile/get-android-device-information.php grab the MAC Address field and try to login with that and ur app password on a computer. Then it might fail in the terminal and then you can see the list of the actual valid IDs. That is what I have to do ONCE if I'm using a new PAAS since most PAAS/IAAS platforms dont give you access to a mac address.

**Comment by tinba12** [comment]
@fergyfresh  I already did that. The stream URL works if you enter it into a browser.

**Comment by RickeyWard** [comment]
I can get it to work, but google flags my account as "suspicious" and the music will only play for 2ish minutes. @tinba12 if you can get the audio stream in the browser and youre getting valid data back from the api, then it's your alexa side. Make sure you have rebuild your model since adding the audio player, and that you're on SSL.

**Comment by fergyfresh** [comment]
Are you using 2FA? This smells like you're not using 2FA.

**Comment by RickeyWard** [comment]
> Are you using 2FA? This smells like you're not using 2FA.

If you're referring to me, I am. You can't create app passwords unless you're using 2FA. I get emails that says "we blocked some suspicious activity on your account"

**Comment by fergyfresh** [comment]
Are you using a device I'd?

**Comment by tinba12** [comment]
Looking through the debug I found this

"request": {
                            "type": "SessionEndedRequest",
                            "requestId": "amzn1.echo-api.request.0f9f08e1-fe1c-4a98-b0b8-96caf1e6ff9f",
                            "timestamp": "2018-12-04T02:11:13Z",
                            "locale": "en-US",
                            "reason": "ERROR",
                            "error": {
                                "type": "INVALID_RESPONSE",
                                "message": "The URL specified in the Play directive is invalid"
                            }
                        }

**Comment by fergyfresh** [comment]
@tinba12 can you get any debug logs from the server (geemusic). the debug logs here are from Alexa.

**Comment by fergyfresh** [comment]
@RickeyWard do this on a terminal on your computer:
Go to this website https://www.myfakeinfo.com/mobile/get-android-device-information.php and get the alphanumeric in the MAC Address field. I'm going to reference that as `android_id` below. 
Whatever your workflow get the pip requirements from this project installed and go into a python shell:
```
from gmusicapi import Mobileclient
email = 'youremail'
password='yourapppassword'
android_id=android_id # i talked about this above.
api = api.login(email, password, android_id)
```
I hope this will fail and give you the valid device ids that are tied to your account. If you then set an environment variable on your server called `ANDROID_ID` to the alphanumeric string you needed to set to login correctly without an exception, you should no longer get the suspicious activity issue.

**Comment by fergyfresh** [comment]
I actually just did it and it's surprisingly simple.

**Comment by everyonce** [comment]
can someone spell out for me how to authorize with a mobile device id?  I've gotten a few steps down the path but I'm not sure if I'm doing the right things...

**Comment by fergyfresh** [comment]
@everyonce does the above comment not work for you? If that's not what you meant, I think in order to get a mobile device id auth'd you have to physically download google play music and login to that app on your phone.

**Comment by RickeyWard** [comment]
I started pulling the latest gmusicapi a couple days ago, but I was not sure what the steps were needed to do an oath signin, though I'm using heroku so getting to a popup redirect is a bit of an issue. Generating a token and putting that in the environment should work.

**Comment by everyonce** [comment]
ok, so I think that's where I'm lost - I've got a phone using gmusic, but how do I use the token(s) from the phone app when I'm running geemusic on my server?

**Comment by fergyfresh** [comment]
@RickeyWard @everyonce I added how to do it via the develop branch of this repo. I also got the tests passing, so its working for me.

**Comment by everyonce** [comment]
ok, I did eventually get it running using that info, tyvm!  However, I was able to confirm the streaming command used isn't allowed for free accounts:
gmusicapi.exceptions.NotSubscribed: Store tracks require a subscription to stream. (https://goo.gl/v1wVHT)

**Comment by tinba12** [comment]
This is what is outputted when I run ask gee music play songs by asia

`03: 15: 36web.1|{
	"context": {
		"AudioPlayer": {
			"playerActivity": "IDLE"
		},
		"System": {
			"apiAccessToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJodHRwczovL2FwaS5hbWF6b25hbGV4YS5jb20iLCJpc3MiOiJBbGV4YVNraWxsS2l0Iiwic3ViIjoiYW16bjEuYXNrLnNraWxsLmQyZTRlNzExLTJjYmEtNDFhNC1iYjNmLTMyZjNkYjJkNGI2YiIsImV4cCI6MTU0NDA2OTczNiwiaWF0IjoxNTQ0MDY2MTM2LCJuYmYiOjE1NDQwNjYxMzYsInByaXZhdGVDbGFpbXMiOnsiY29uc2VudFRva2VuIjpudWxsLCJkZXZpY2VJZCI6ImFtem4xLmFzay5kZXZpY2UuQUZRUUxUM0RPRUpVUjRYSElNSERNUFY3TlhYWU1FSjJYNDdLRVVQRzJZSjY1TUdLSU9UQ1BUR0paTFg3VktHWVI2RTVYTTc1VTNTNEE3NzM3WEQ3NjdOVUJSS0pUWEZYR1E3N1FGUklRQTVCUEdWMkJZWkQ1VFZKVUdBNkFBVVZJN1BVU1NPWU1BQ0NZWDdUTlczSkFOSlZUTUtHTUlFT0dJSTIzM0xWWlJGTTdKQ09YSU5BQSIsInVzZXJJZCI6ImFtem4xLmFzay5hY2NvdW50LkFGS1BaTlNYRUQzWlMyNlQzNzJXWklaU0pQQ1dIQ1NXRks0MjJKVjZBNk02N0xJRzVQT0s0TDROVk1XSVZHNzVGNktHVFhHQUJUT0tGNUszUVVLSlNZUldJUFpITlFURUFMQzREVVJZWFBSU0xJRFFMRlhBNUNUWVdKN0RCN0tPRVpDSkg3VVlVTEVaUzdRRURSRlJYQ0syQ1UzTFJCTVlOSENKQkRUNUlDUURFNDJDVEdSRVk2M0tTWUNIR0g0N0dGWEEyRlNLU0dQWFlaSSJ9fQ.L575VXj81tDUSdOwr-Nqh1v4botYMBZXHOzTNyNu1WsgElaLj4KQo3_F1nS04vT7lpM-zOiApNhZzBME7OCkUhSO-HnNFg6zx0w28queZifZFp9OxQKINIilchOsGN0Anih3s8G2IK6K020thDRB7PpIFVieisyqlZcjW7phJ1qKvkXy6URairsw6yXg0rfWSgFBxrFRq_3_hNdJKQqQAXaq3vp7vqMnvXLkL190dA-6KMJUiDNCp5S3du1yvrUU7IB03mDO9dxX5U3Q4DnFymvi787mIGbhP6PRkcZgOU1Is-NOLLRlTxh5kIhA7vaCU0AUqQ_z1GJx0KPui6EZtQ",
			"apiEndpoint": "https://api.amazonalexa.com",
			"application": {
				"applicationId": "amzn1.ask.skill.d2e4e711-2cba-41a4-bb3f-32f3db2d4b6b"
			},
			"device": {
				"deviceId": "amzn1.ask.device.AFQQLT3DOEJUR4XHIMHDMPV7NXXYMEJ2X47KEUPG2YJ65MGKIOTCPTGJZLX7VKGYR6E5XM75U3S4A7737XD767NUBRKJTXFXGQ77QFRIQA5BPGV2BYZD5TVJUGA6AAUVI7PUSSOYMACCYX7TNW3JANJVTMKGMIEOGII233LVZRFM7JCOXINAA",
				"supportedInterfaces": {
					"AudioPlayer": {
						
					}
				}
			},
			"user": {
				"userId": "amzn1.ask.account.AFKPZNSXED3ZS26T372WZIZSJPCWHCSWFK422JV6A6M67LIG5POK4L4NVMWIVG75F6KGTXGABTOKF5K3QUKJSYRWIPZHNQTEALC4DURYXPRSLIDQLFXA5CTYWJ7DB7KOEZCJH7UYULEZS7QEDRFRXCK2CU3LRBMYNHCJBDT5ICQDE42CTGREY63KSYCHGH47GFXA2FSKSGPXYZI"
			}
		},
		"Viewport": {
			"currentPixelHeight": 600,
			"currentPixelWidth": 1024,
			"dpi": 160,
			"experiences": [{
				"arcMinuteHeight": 144,
				"arcMinuteWidth": 246,
				"canResize": false,
				"canRotate": false
			}],
			"pixelHeight": 600,
			"pixelWidth": 1024,
			"shape": "RECTANGLE",
			"touch": ["SINGLE"]
		}
	},
	"request": {
		"intent": {
			"confirmationStatus": "NONE",
			"name": "GeeMusicPlayArtistIntent",
			"slots": {
				"artist_name": {
					"confirmationStatus": "NONE",
					"name": "artist_name",
					"source": "USER",
					"value": "Asia"
				}
			}
		},
		"locale": "en-US",
		"requestId": "amzn1.echo-api.request.79436b2c-cc72-4e9f-84cf-b10ba0438948",
		"timestamp": "2018-12-06T03:15:36Z",
		"type": "IntentRequest"
	},
	"session": {
		"application": {
			"applicationId": "amzn1.ask.skill.d2e4e711-2cba-41a4-bb3f-32f3db2d4b6b"
		},
		"new": true,
		"sessionId": "amzn1.echo-api.session.22fe3c16-3238-487f-889b-c58e8abd47c5",
		"user": {
			"userId": "amzn1.ask.account.AFKPZNSXED3ZS26T372WZIZSJPCWHCSWFK422JV6A6M67LIG5POK4L4NVMWIVG75F6KGTXGABTOKF5K3QUKJSYRWIPZHNQTEALC4DURYXPRSLIDQLFXA5CTYWJ7DB7KOEZCJH7UYULEZS7QEDRFRXCK2CU3LRBMYNHCJBDT5ICQDE42CTGREY63KSYCHGH47GFXA2FSKSGPXYZI"
		}
	},
	"version": "1.0"
}03: 15: 36web.1|{
	"offsetInMilliseconds": 0,
	"playerActivity": "IDLE",
	"token": "cdb7b36e-4ce8-4a46-b40a-4d7030362c76",
	"url": "https:///<server>/alexa/stream/Twthf5iczxdtsk34jacsa43ktpm"
}03: 15: 37web.1|{
	"response": {
		"card": {
			"image": {
				"largeImageUrl": "https://lh3.googleusercontent.com/JgVIgIQSzdJ6ViETQE--4h51BpwwBW4tIdI9rsOT0FGrm8vujWe6hH2pG-eItLJj0IODueZ16Q",
				"smallImageUrl": "https://lh3.googleusercontent.com/JgVIgIQSzdJ6ViETQE--4h51BpwwBW4tIdI9rsOT0FGrm8vujWe6hH2pG-eItLJj0IODueZ16Q"
			},
			"text": "",
			"title": "Playing top tracks by Asia",
			"type": "Standard"
		},
		"directives": [{
			"audioItem": {
				"stream": {
					"offsetInMilliseconds": 0,
					"token": "bbd950de-c35f-4b23-a090-e9b3b5500ec7",
					"url": "https:///<server>/alexa/stream/Twthf5iczxdtsk34jacsa43ktpm"
				}
			},
			"playBehavior": "REPLACE_ALL",
			"type": "AudioPlayer.Play"
		}],
		"outputSpeech": {
			"text": "Playing top tracks by Asia",
			"type": "PlainText"
		},
		"shouldEndSession": true
	},
	"sessionAttributes": {
		
	},
	"version": "1.0"
}03: 15: 37web.1|72.21.217.96--[06/Dec/201803: 15: 37]"POST /alexa HTTP/1.1"200-03: 15: 38web.1|{
	"context": {
		"AudioPlayer": {
			"playerActivity": "IDLE"
		},
		"System": {
			"apiAccessToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjEifQ.eyJhdWQiOiJodHRwczovL2FwaS5hbWF6b25hbGV4YS5jb20iLCJpc3MiOiJBbGV4YVNraWxsS2l0Iiwic3ViIjoiYW16bjEuYXNrLnNraWxsLmQyZTRlNzExLTJjYmEtNDFhNC1iYjNmLTMyZjNkYjJkNGI2YiIsImV4cCI6MTU0NDA2OTczOCwiaWF0IjoxNTQ0MDY2MTM4LCJuYmYiOjE1NDQwNjYxMzgsInByaXZhdGVDbGFpbXMiOnsiY29uc2VudFRva2VuIjpudWxsLCJkZXZpY2VJZCI6ImFtem4xLmFzay5kZXZpY2UuQUZRUUxUM0RPRUpVUjRYSElNSERNUFY3TlhYWU1FSjJYNDdLRVVQRzJZSjY1TUdLSU9UQ1BUR0paTFg3VktHWVI2RTVYTTc1VTNTNEE3NzM3WEQ3NjdOVUJSS0pUWEZYR1E3N1FGUklRQTVCUEdWMkJZWkQ1VFZKVUdBNkFBVVZJN1BVU1NPWU1BQ0NZWDdUTlczSkFOSlZUTUtHTUlFT0dJSTIzM0xWWlJGTTdKQ09YSU5BQSIsInVzZXJJZCI6ImFtem4xLmFzay5hY2NvdW50LkFGS1BaTlNYRUQzWlMyNlQzNzJXWklaU0pQQ1dIQ1NXRks0MjJKVjZBNk02N0xJRzVQT0s0TDROVk1XSVZHNzVGNktHVFhHQUJUT0tGNUszUVVLSlNZUldJUFpITlFURUFMQzREVVJZWFBSU0xJRFFMRlhBNUNUWVdKN0RCN0tPRVpDSkg3VVlVTEVaUzdRRURSRlJYQ0syQ1UzTFJCTVlOSENKQkRUNUlDUURFNDJDVEdSRVk2M0tTWUNIR0g0N0dGWEEyRlNLU0dQWFlaSSJ9fQ.M0Gapzxrg00ayRh7_6djYS00iYaGP1sf4bEg1r5eWx29yv5TgwW_JbN1UUpR6mp6b7IuTc0u6FoTpryANwxbZi_itLO-Fud6ERpDnXxVs4Sg_UXQ2aA8yh5VEUB3--lq1MI0cdjabiBwvDM9pj9QOB4gZYZjZc0OIAJUUdXwPNSEuG550NDsIss-au4IX-KljtIF4-IZBr8fdU3OiFtUiZC0HLZRWU2KUcZY3NMdPSgMPL8UQuSG0ig0GsoBsu3zdd_xduB9iEfKxFfz8FTaaVmtt4rc9Yicg-nF7RvBbz3n8sgLfqN4Y-t55vqBaEmgRCXo6T6deNdizbqPuD5mzA",
			"apiEndpoint": "https://api.amazonalexa.com",
			"application": {
				"applicationId": "amzn1.ask.skill.d2e4e711-2cba-41a4-bb3f-32f3db2d4b6b"
			},
			"device": {
				"deviceId": "amzn1.ask.device.AFQQLT3DOEJUR4XHIMHDMPV7NXXYMEJ2X47KEUPG2YJ65MGKIOTCPTGJZLX7VKGYR6E5XM75U3S4A7737XD767NUBRKJTXFXGQ77QFRIQA5BPGV2BYZD5TVJUGA6AAUVI7PUSSOYMACCYX7TNW3JANJVTMKGMIEOGII233LVZRFM7JCOXINAA",
				"supportedInterfaces": {
					"AudioPlayer": {
						
					}
				}
			},
			"user": {
				"userId": "amzn1.ask.account.AFKPZNSXED3ZS26T372WZIZSJPCWHCSWFK422JV6A6M67LIG5POK4L4NVMWIVG75F6KGTXGABTOKF5K3QUKJSYRWIPZHNQTEALC4DURYXPRSLIDQLFXA5CTYWJ7DB7KOEZCJH7UYULEZS7QEDRFRXCK2CU3LRBMYNHCJBDT5ICQDE42CTGREY63KSYCHGH47GFXA2FSKSGPXYZI"
			}
		},
		"Viewport": {
			"currentPixelHeight": 600,
			"currentPixelWidth": 1024,
			"dpi": 160,
			"experiences": [{
				"arcMinuteHeight": 144,
				"arcMinuteWidth": 246,
				"canResize": false,
				"canRotate": false
			}],
			"pixelHeight": 600,
			"pixelWidth": 1024,
			"shape": "RECTANGLE",
			"touch": ["SINGLE"]
		}
	},
	"request": {
		"error": {
			"message": "The URL specified in the Play directive is invalid",
			"type": "INVALID_RESPONSE"
		},
		"locale": "en-US",
		"reason": "ERROR",
		"requestId": "amzn1.echo-api.request.285702af-416b-4820-b373-9b15bcf7ece1",
		"timestamp": "2018-12-06T03:15:38Z",
		"type": "SessionEndedRequest"
	},
	"session": {
		"application": {
			"applicationId": "amzn1.ask.skill.d2e4e711-2cba-41a4-bb3f-32f3db2d4b6b"
		},
		"new": false,
		"sessionId": "amzn1.echo-api.session.22fe3c16-3238-487f-889b-c58e8abd47c5",
		"user": {
			"userId": "amzn1.ask.account.AFKPZNSXED3ZS26T372WZIZSJPCWHCSWFK422JV6A6M67LIG5POK4L4NVMWIVG75F6KGTXGABTOKF5K3QUKJSYRWIPZHNQTEALC4DURYXPRSLIDQLFXA5CTYWJ7DB7KOEZCJH7UYULEZS7QEDRFRXCK2CU3LRBMYNHCJBDT5ICQDE42CTGREY63KSYCHGH47GFXA2FSKSGPXYZI"
		}
	},
	"version": "1.0"
}03: 15: 38web.1|{
	"offsetInMilliseconds": 0,
	"playerActivity": "IDLE",
	"token": "bbd950de-c35f-4b23-a090-e9b3b5500ec7",
	"url": "https:///<server>/alexa/stream/Twthf5iczxdtsk34jacsa43ktpm"
}03: 15: 38web.1|72.21.217.96--[06/Dec/201803: 15: 38]"POST /alexa HTTP/1.1"200-`

**Comment by tinba12** [comment]
This is the log when i run ask gee music play songs by asia
[debug.txt](https://github.com/stevenleeg/geemusic/files/2651579/debug.txt)

**Comment by fergyfresh** [comment]
What do you have as the callback url in the Alexa skills console? `APP_URL + '/alexa'` is what it should be.

**Comment by steveseguin** [comment]
My god. It works. (finally, lol). 👍 

I had several device_ids listed after that "_python -c 'from gmusicapi import Mobileclient ..._" step, some long and some short. Using the last device_id in the list, a long one, didn't work for me-- I got a forbidden error when trying to play a song with it.  I then went thru the steps again and used a short device_id; for whatever reason this time it worked.

Fantastic. I am using the development branch, ubuntu, python 3.6, and paid Google Music.  

Thank you.
